focus/operations/src/maintenance/mod.rs

```diff
@@ -328,22 +346,106 @@ impl Runner<'_> {
     fn run_in_path(&self, time_period: TimePeriod, path: &Path) -> Result<()> {
         info!(?time_period, ?path, "running tasks");
         set_default_git_maintenance_config(path)?;
+
+        let maint_result_metric = "maint_result".to_string();
         match self.run_maint(time_period, path) {
             Ok(MaintResult::Success(status)) => {
                 if status.success() {
                     debug!(?time_period, ?path, "completed maintenance",)
                 } else {
                     warn!(?path, exit_status = ?status, "maintenance failed");
+                    let mut maint_custom_map = HashMap::new();
+                    maint_custom_map.insert(
+                        maint_result_metric,
+                        self.maint_exitstatus_metric_helper(status),
+                    );
+                    self.add_ti_invocation_message(&maint_custom_map);
                 }
             }
-            Ok(MaintResult::LockFailed) => warn!(?path, "failed to acquire lock"),
+            Ok(MaintResult::LockFailed) => {
+                warn!(?path, "failed to acquire lock");
+                let maint_custom_map =
+                    HashMap::from([(maint_result_metric, "lock_failed".to_string())]);
+                self.add_ti_invocation_message(&maint_custom_map);
+            }
             Err(e) => {
-                warn!(?path, ?e, "failed runing git-maintenance");
+                warn!(?path, ?e, "failed running git-maintenance");
+                let maint_custom_map =
+                    HashMap::from([(maint_result_metric, "git_error".to_string())]);
+                self.add_ti_invocation_message(&maint_custom_map);
             }
         }
         Ok(())
     }
 
+    fn add_ti_invocation_message(&self, maint_custom_map: &HashMap<String, String>) {
+        let ti_client = self.app.tool_insights_client();
+        ti_client.get_inner().add_invocation_message(
+            SystemTime::now(),
+            None,
+            Some(maint_custom_map),
+        );
+    }
+
+    fn maint_exitstatus_metric_helper(&self, exit_status: ExitStatus) -> String {
```

Comment:
```suggestion
    fn maint_exit_status_metric_helper(&self, exit_status: ExitStatus) -> String {
```

Name this `*_exit_status_*`;  when converting `CamelCase` to `snake_case`, you should put an underscore in between each word.

Reply:
Fixed.